lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -70,7 +76,19 @@ def execute(self):
                         )
 
             if tool_info:
-                self.print_results(tool_info, None, self.context.with_option)
+                found = self.print_results(tool_info, self.context.with_option)
+                if not found:
+                    self.logger.warn(
+                        "No tools found%s",
+                        f' in group "{self.context.group[0]}"'
+                        if self.context.group
+                        else "",
+                    )
```

Comment:
The combination of logger with f-string syntax always bugs me. The lazy evaluation isn't an issue here since we presumably don't expect `WARN` to be disabled, but it still looks odd. (Even aside from the blackened ternary.)

Reply:
Yeah, when I saw what Black did to it, I almost ripped it back out.  (In an earlier iteration, the ternary all fit on one line, so it was reasonable, but then the indentation level got increased or something, and then Black wrecked it.)

I'll try building the string before the logger invocation.

Reply:
OK, I like that better:
```
                    msg = "No tools found"
                    if self.context.group:
                        msg += f' in group "{self.context.group[0]}"'
                    self.logger.warn(msg)
```
(I think I'm going to give up using the ternary in Python. 😢)